### ISSUE TITLE
[Feature] Allow tests to be written in typescript with `"type": "module"` in the absence of a config file

### ISSUE THREAD

**Issue by lilnasy** [microsoft/playwright#28534]
In the process of moving from `vitest` to `@playwright/test`, I kept running into this issue:

```bash
> pnpm playwright test
TypeError: Unknown file extension ".ts" for /workspace/project/tests/basics.spec.ts



Error: No tests found




```

After browsing around and finding https://github.com/microsoft/playwright/issues/27827 and https://github.com/microsoft/playwright/issues/23662 to no help, I realised it was because I did not have playwright.config in my project. I do not want to change any defaults and I certainly don't want another root-level config if I can avoid it.

To reproduce, run `npm init playwright` with all default options, then remove playwright.config file and add `"type": "module"` to the `package.json`.

**Comment by dgozman** [comment]
@lilnasy Unfortunately, Playwright is not going to support this in the near future. To work with typescript and `"type": "module"`, we need to engage the ESM loader, and we do so when we encounter config that is a module itself. Otherwise, we would have to engage the ESM loader upon reading your test files, which is too late. That said, once ESM loader story in Node.js matures, there might be changes in this area.

**Comment by mxschmitt** [comment]
This might be fixed in v1.43.

**Comment by lilnasy** [comment]
Yes, it has!

Thanks @pavelfeldman for implementing ts loader on top of newer node APIs. Thanks @dgozman for flipping the switch. Thanks @mxschmitt for the shout.

- Fixed by #29439 followed by #29991.
